### PR TITLE
Update _Muscle.py

### PR DESCRIPTION
**Pull request by JoshuaMeyers**
Added missing flags from MUSCLE 3.8 including -gapextend and -matrix.

### CODE REVIEW

**Comment by peterjc** [comment]
Thanks for working on this this. At first reading that looks good, except that TravisCI is failing on one of the style checks we've added recently:

```
flake8 Bio/
Bio/Align/Applications/_Muscle.py:168:84: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:170:86: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:172:90: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:276:89: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:303:91: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:304:87: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:455:90: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:500:89: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:501:88: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:502:93: W291 trailing whitespace
Bio/Align/Applications/_Muscle.py:520:90: W291 trailing whitespace
```

This is at least easy to fix. You can install flake8 locally with ``pip install flake8`` which should help with checking this.

Did ``-maxdiagbreak`` get renamed in Muscle? If so then a comment is probably enough here... although it would make life awkward for anyone using this parameter via Biopython.

**Comment by JoshuaMeyers** [comment]
- Style changes made to agree with Travis CI
- readded -maxdiagbreak as a legacy option (it was deprecated in Muscle 3.7).
(marked with a comment for deprecation in the future)

File: Bio/Align/Applications/_Muscle.py [hunk 1]
```diff
@@ -22,9 +22,10 @@ class MuscleCommandline(AbstractCommandline):
     >>> muscle_exe = r"C:\Program Files\Aligments\muscle3.8.31_i86win32.exe"
     >>> in_file = r"C:\My Documents\unaligned.fasta"
     >>> out_file = r"C:\My Documents\aligned.fasta"
-    >>> muscle_cline = MuscleCommandline(muscle_exe, input=in_file, out=out_file)
+    >>> muscle_cline=MuscleCommandline(muscle_exe,input=in_file,out=out_file)
```
Comment (peterjc) [inline Bio/Align/Applications/_Muscle.py]: Those spaces should be there as per the PEP8 style guidelines.

File: Bio/Align/Applications/_Muscle.py [hunk 2]
```diff
@@ -22,9 +22,10 @@ class MuscleCommandline(AbstractCommandline):
...
     >>> print(muscle_cline)
-    "C:\Program Files\Aligments\muscle3.8.31_i86win32.exe" -in "C:\My Documents\unaligned.fasta" -out "C:\My Documents\aligned.fasta"
+    "C:\Program Files\Aligments\muscle3.8.31_i86win32.exe"
+    -in "C:\My Documents\unaligned.fasta" -out "C:\My Documents\aligned.fasta"
```
Comment (peterjc) [inline Bio/Align/Applications/_Muscle.py]: It looks like you've inserted a line break. Unfortunately that will break the doctest, so in this situation we just have to accept some occasionally long lines.

File: Bio/Align/Applications/_Muscle.py [hunk 3]
```diff
@@ -72,275 +75,353 @@ def __init__(self, cmd="muscle", **kwargs):
...
+            #                                              diagonals that
+            #                                              allows them to
+            #                                              merge into one
+            #                                              diagonal.
+            _Option(["-diagbreak", "diagbreak"],
+                    "Maximum distance between two diagonals that allows "
+                    "them to merge into one diagonal",
+                    checker_function=lambda x: isinstance(x, int),
+                    equate=False),
+            _Option(["-maxdiagbreak", "maxdiagbreak"],      # deprecated 3.8
```
Comment (peterjc) [inline Bio/Align/Applications/_Muscle.py]: I would include something like this in the docstring for this attribute: ``"Deprecated in v3.8, use -diagbreak instead."``

File: Bio/Align/Applications/_Muscle.py (hunk 2 above)
Comment (peterjc) [inline Bio/Align/Applications/_Muscle.py]: This line break is the cause of the TravisCI test failure, e.g.:

```
======================================================================
FAIL: MuscleCommandline (Bio.Align.Applications._Muscle)
Doctest: Bio.Align.Applications._Muscle.MuscleCommandline
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/python/3.4.2/lib/python3.4/doctest.py", line 2193, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for Bio.Align.Applications._Muscle.MuscleCommandline
  File "/home/travis/build/biopython/biopython/build/lib.linux-x86_64-3.4/Bio/Align/Applications/_Muscle.py", line 13, in MuscleCommandline
----------------------------------------------------------------------
File "/home/travis/build/biopython/biopython/build/lib.linux-x86_64-3.4/Bio/Align/Applications/_Muscle.py", line 26, in Bio.Align.Applications._Muscle.MuscleCommandline
Failed example:
    print(muscle_cline)
Expected:
    "C:\Program Files\Aligments\muscle3.8.31_i86win32.exe"
    -in "C:\My Documents\unaligned.fasta" -out "C:\My Documents\aligned.fasta"
Got:
    "C:\Program Files\Aligments\muscle3.8.31_i86win32.exe" -in "C:\My Documents\unaligned.fasta" -out "C:\My Documents\aligned.fasta"
----------------------------------------------------------------------
Ran 271 tests in 204.323 seconds
```

**Comment by JoshuaMeyers** [comment]
- Deprecation comment added to docstring of -maxdiagbreak
- Spaces added and linebreak removed to comply with PEP 8 style guide

**Comment by peterjc** [comment]
Doctest are painfully fragile, it now says:

```
======================================================================
FAIL: MuscleCommandline (Bio.Align.Applications._Muscle)
Doctest: Bio.Align.Applications._Muscle.MuscleCommandline
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/python/2.7.9/lib/python2.7/doctest.py", line 2226, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for Bio.Align.Applications._Muscle.MuscleCommandline
  File "/home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/Align/Applications/_Muscle.py", line 13, in MuscleCommandline
----------------------------------------------------------------------
File "/home/travis/build/biopython/biopython/build/lib.linux-x86_64-2.7/Bio/Align/Applications/_Muscle.py", line 26, in Bio.Align.Applications._Muscle.MuscleCommandline
Failed example:
    print(muscle_cline)
Expected:
    "C:\Program Files\Aligments\muscle3.8.31_i86win32.exe" -in "C:\My Documents\unaligned.fasta" -out "C:\My Documents\aligned.fasta"
    You would typically run the command line with muscle_cline() or via
    the Python subprocess module, as described in the Biopython tutorial.
Got:
    "C:\Program Files\Aligments\muscle3.8.31_i86win32.exe" -in "C:\My Documents\unaligned.fasta" -out "C:\My Documents\aligned.fasta"
----------------------------------------------------------------------
```

The problem is you've accidentally deleted the blank line at the end of the doctest.

If you want to reproduce this on your machine, rather than running the full test suite, you can do just:

```
$ python setup.py install
$ cd Tests
$ python run_tests.py doctest
```

(see also #531)

**Comment by JoshuaMeyers** [comment]
Thanks for all the useful feedback. Very useful learning for me.
The source of the problem here is that I haven't been able to run the doctest locally. It says my directory isn't within the biopython source tree.  (I am using 1.68 up to date installed from conda)

**Comment by peterjc** [comment]
Thank you - this looks ready now.

Are you happy to dual licence your contributions under both the _Biopython License Agreement_ AND the _BSD 3-Clause License_? See #898.

Also, are you happy to be thanked by name in the ``NEWS`` and ``CONTRIB`` files?

I'm note quite sure how you are combining Biopython from conda with Biopython from git, but that does likely explain why you've had trouble running the tests locally. Thankfully TravisCI is a useful fallback (although it takes a while to run the full test suite).

**Comment by JoshuaMeyers** [comment]
I am combining them rather manually I'll admit...

Yes I am happy with the dual license and to be thanked by name (do I need to do anything?)

Yep the checks are very thorough!

Thanks again

**Comment by peterjc** [comment]
Thank you, that's done.

**Pull request merged**